### ISSUE TITLE
[BUG] Lua error

### ISSUE THREAD

**Issue by TheLostInPlace** [teelolws/TLDRMissions#214]
```2x TLDRMissions\simulation.lua:891: attempt to index local 'buff' (a nil value)
[string "@TLDRMissions\simulation.lua"]:891: in function `unregisterPassive'
[string "@TLDRMissions\simulation.lua"]:483: in function <TLDRMissions\simulation.lua:474>
[string "@TLDRMissions\simulation.lua"]:549: in function <TLDRMissions\simulation.lua:516>
[string "@TLDRMissions\simulation.lua"]:777: in function <TLDRMissions\simulation.lua:700>
[string "@TLDRMissions\simulation.lua"]:1155: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function `turnRecursion'
[string "@TLDRMissions\simulation.lua"]:1171: in function <TLDRMissions\simulation.lua:23>
[string "@TLDRMissions\simulation.lua"]:1288: in function `Simulate'
[string "@TLDRMissions\combinations.lua"]:7: in function `func'
[string "@TLDRMissions\workhandler.lua"]:50: in function <TLDRMissions\workhandler.lua:23>

Locals:
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1 = <table> {
 }
}
(for control) = 1
_ = 1
target = <table> {
 maxHP = 620
 spells = <table> {
 }
 autoAttack = <table> {
 }
 boardIndex = 4
 name = "Ardenweald Grovetender"
 followerID = "0xFFFFFFFFFFFFFFFF"
 baseAttack = 155
 buffs = <table> {
 }
 HP = 243
 thorns = <table> {
 }
}
buff = nil
(*temporary) = "Invigorating Herbs4"
(*temporary) = "4"
(*temporary) = <table> {
 1 = <table> {
 }
}
(*temporary) = "attempt to index local 'buff' (a nil value)"
getTargets = <function> defined @TLDRMissions\simulation.lua:108
follower = <table> {
 maxHP = 1000
 spells = <table> {
 }
 isDead = true
 unregisterPassive = <function> defined @TLDRMissions\simulation.lua:888
 autoAttack = <table> {
 }
 boardIndex = 0
 name = "Niya"
 followerID = "0x0000000037A97D71"
 baseAttack = 200
 buffs = <table> {
 }
 HP = -62
 thorns = <table> {
 }
}
spell = <table> {
 effects = <table> {
 }
 onCooldown = 0
 duration = 0
 cooldown = 0
 spellID = 90
}
```

Errors out on mission Master and Pupil: https://i.imgur.com/ZNyGypD.png

**Comment by TheLostInPlace** [comment]
```3x TLDRMissions\simulation.lua:891: attempt to index local 'buff' (a nil value)
[string "@TLDRMissions\simulation.lua"]:891: in function `unregisterPassive'
[string "@TLDRMissions\simulation.lua"]:483: in function <TLDRMissions\simulation.lua:474>
[string "@TLDRMissions\simulation.lua"]:549: in function <TLDRMissions\simulation.lua:516>
[string "@TLDRMissions\simulation.lua"]:777: in function <TLDRMissions\simulation.lua:700>
[string "@TLDRMissions\simulation.lua"]:1155: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function <TLDRMissions\simulation.lua:1090>
[string "@TLDRMissions\simulation.lua"]:1164: in function `turnRecursion'
[string "@TLDRMissions\simulation.lua"]:1171: in function <TLDRMissions\simulation.lua:23>
[string "@TLDRMissions\simulation.lua"]:1288: in function `Simulate'
[string "@TLDRMissions\combinations.lua"]:7: in function `func'
[string "@TLDRMissions\workhandler.lua"]:50: in function <TLDRMissions\workhandler.lua:23>

Locals:
(for generator) = <function> defined =[C]:-1
(for state) = <table> {
 1 = <table> {
 }
}
(for control) = 1
_ = 1
target = <table> {
 maxHP = 620
 spells = <table> {
 }
 autoAttack = <table> {
 }
 boardIndex = 4
 name = "Ardenweald Grovetender"
 followerID = "0xFFFFFFFFFFFFFFFF"
 baseAttack = 155
 buffs = <table> {
 }
 HP = 243
 thorns = <table> {
 }
}
buff = nil
(*temporary) = "Invigorating Herbs4"
(*temporary) = "4"
(*temporary) = <table> {
 1 = <table> {
 }
}
(*temporary) = "attempt to index local 'buff' (a nil value)"
getTargets = <function> defined @TLDRMissions\simulation.lua:108
follower = <table> {
 maxHP = 1000
 spells = <table> {
 }
 isDead = true
 unregisterPassive = <function> defined @TLDRMissions\simulation.lua:888
 autoAttack = <table> {
 }
 boardIndex = 0
 name = "Niya"
 followerID = "0x0000000037A97D71"
 baseAttack = 200
 buffs = <table> {
 }
 HP = -62
 thorns = <table> {
 }
}
spell = <table> {
 effects = <table> {
 }
 onCooldown = 0
 duration = 0
 cooldown = 0
 spellID = 90
}
```

Also happening on Flight of the Necrorays. Using latest version of TLDR that allows for max simulations.

**Comment by TheLostInPlace** [comment]
Had to send adventurers on a "doomed" run on those two missions so the addon would stop failing when calculating those two (thus bricking itself until next reload, to then brick again ad infinitum)

**Comment by teelolws** [comment]
Yeah I found the problem. Patch going up asap.

**Comment by teelolws** [comment]
Thanks for the report. Fixed in 9.2-059.
